lib/stack.js

```diff
@@ -0,0 +1,33 @@
+export class Stack {
+    constructor() {
+      this.top = -1;
+      this.items = {};
+  }
+  
+    push(value)
+    {
+      this.top++;
+      this.items[this.top] = value;
+    } 
+  
+    pop() { 
+      this.top--;
+      delete this.items[this.top]
+    }
+  
+    getLength(){
+      return Object.keys(this.items).length;
+    }
+
+    peek(){
+      return this.items[this.top];
+    }
+
+    clear(){
+      for (let item in Object.keys(this.items)){
+        delete this.items[item];
+      };
```

Comment:
Loops can be expensive to run, so when we write a loop, the reason should be that each item in the loop has a different input and a different output that we want to preserve. In our case, we are just deleting the whole thing and we do not care about individually processing the items in the array.

I would suggest that we do this in the simplest way possible, which is to reset the properties to what they are in the constructor:

```
clear(){
  this.items = {};
  this.top = -1;
  this.length = 0;
}
```

OR better yet, to reduce code duplication, we can put these property resets into a method:

```
constructor(){
  this.reset()
}


reset() {
  this.items = {};
  this.top = -1;
  this.length = 0;
}
```

Reply:
Ah yes, this is so simple and natural, can't believe I haven't thought about it!! O.o

In case, with 
```js
constructor(){
  this.reset()
}

reset() {
  this.items = {};
  this.top = -1;
  this.length = 0;
}
```
- that is a bit confusing to me having this method in the constructor from reading and understanding what's going on point of view. Is that a preferred way do it?

Reply:
Yes having a `reset()` or `init()` method is preferred over duplicated logic:
```
constructor(){
   this.items = {};
  this.top = -1;
  this.length = 0;
}

reset() {
  this.items = {};
  this.top = -1;
  this.length = 0;
}
```

You could rename the method to whatever you think reads the best. `init()` is probably going to be the most familiar to other devs